docs/reference/api-gateway/http.md

```diff
@@ -0,0 +1,171 @@
+        "authenticationMode": "none",
+        "upstreams": [
+            {
+                "kind": "nucliofunction",
+                "nucliofunction": {
+                    "name": "function-name-to-invoke",
+                },
+                "percentage": 0
+            }
+        ],
+        "host": "<apigateway-name>-<project-name>.<nuclio-host-name>"
+    },
+    "metadata": {
+        "labels": {
+            "nuclio.io/project-name": "default"
+        },
+        "name": "<apigateway-name>"
+    }
+}
+```
+
+<a id="invoke-none"></a>
+### Invoke API Gateways
+
+To invoke it, simply send GET/POST request to `<apigateway-name>-<project-name>.<nuclio-host-name>` (aka - `spec.host` )
+If your function accepts a request body, you can send it as a request body.
+
+<a id="basic-auth"></a>
+## Basic Authentication
+
+Basic authentication is a way to authenticate users by providing a username and password.
+
+
+<a id="create-basic"></a>
+### Create API Gateways
+
+You can create an api gateway with basic authentication by sending a POST request to the following endpoint:
+
+```
+<nuclio-host-name>/api/api_gateways
+```
+With the following request body:
+```json
+{
+    "spec": {
+        "name": "<apigateway-name>",
+        "description": "some description",
+        "path": "/some/path",
+        "authenticationMode": "basicAuth",
+        "upstreams": [
+            {
+                "kind": "nucliofunction",
+                "nucliofunction": {
+                    "name": "function-name-to-invoke"
+                },
+                "percentage": 0
+            }
+        ],
+        "host": "<apigateway-name>-<project-name>.<nuclio-host-name>",
+        "authentication": {
+            "basicAuth": {
+                "username": "some-username",
+                "password": "some-password"
+            }
+        }
+    },
+    "metadata": {
+        "labels": {
+            "nuclio.io/project-name": "default"
+        },
+        "name": "<apigateway-name>"
+    }
+}
+```
+
+<a id="invoke-basic"></a>
+### Invoke API Gateways
+
+To invoke it, send GET/POST request to `<apigateway-name>-<project-name>.<nuclio-host-name>` (aka - `spec.host` )
+with the following header:
+```
+key: Authorization
+value: Basic base64encode("username:password")
+```
+
+*Example*:
+
+For the following credentials:
+```
+"username": "some-username"
+"password": "some-password"
+```
+base64 encoding of "some-username:some-password" is `c29tZS11c2VybmFtZTpzb21lLXBhc3N3b3Jk`, so the resulting header is:
+```
+Authorization: Basic c29tZS11c2VybmFtZTpzb21lLXBhc3N3b3Jk
+```
+
+Invoking the function without the above header will respond with `401 Authorization Required`
+
+<a id="canary-function"></a>
+## Canary Function
+
+You can add a canary function to an api gateway by adding another upstream to the api gateway.
+
+You can control the percentage of traffic that goes to the canary function by changing the percentage of the upstream.
```

Comment:
switch for relevancy

Reply:
switch what?..

Reply:
switch lines, sorry :) line 135 contents should come first